### PR TITLE
UICAT-100 Edit authority record

### PR DESCRIPTION
**Pull request by EstebanSM**
## Purpose

[UICAT-100 Edit authority record](https://issues.folio.org/browse/UICAT-100)

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by matteopascucci — APPROVED** [review]

File: src/components/Cataloguing/Marc/Record/index.js [hunk 1]
```diff
@@ -336,11 +361,11 @@ class Record extends React.Component<
 
   deleteRecord = async () => {
     let { statusCode } = this.state;
-    const { recordDetail, store, router, reset } = this.props;
+    const { recordDetail, store, router, reset, translate, toggleFilterPane, data: { search: { segment } } } = this.props;
     await del(
       buildUrl(
         store.getState(),
-        C.ENDPOINT.BIBLIOGRAPHIC_RECORD + '/' + recordDetail.id,
+        segment === C.SEARCH_SEGMENT.BIBLIOGRAPHIC ? C.ENDPOINT.BIBLIOGRAPHIC_RECORD + '/' + recordDetail.id : C.ENDPOINT.AUTHORITY_RECORD + '/' + recordDetail.id,
```
Comment (ryandberger) [inline src/components/Cataloguing/Marc/Record/index.js:368]: I recommend using string interpolation here to make this easier to read.

File: src/components/Cataloguing/Marc/Record/index.js [hunk 2]
```diff
@@ -457,22 +484,26 @@ class Record extends React.Component<
...
-    const { leaderData } = this.props;
+    const { leaderData, translate, data: { search: { segment } } } = this.props;
     const { isEditMode, id } = this.state;
     const bibliographicRecord = this.getCurrentRecord();
 
+    const childProps = { ...this.props };
+    delete childProps.emptyRecord;
+    delete childProps.emptyRecordAuth;
+
+    const paneTitle = segment === C.SEARCH_SEGMENT.BIBLIOGRAPHIC
```
Comment (ryandberger) [inline src/components/Cataloguing/Marc/Record/index.js:495]: This should be simplified. You could just test for edit mode, then have other conditions in an ELSE clause for example.

File: src/components/Search/Panel/SearchPanel.js [hunk 3]
```diff
@@ -41,36 +41,46 @@ class SearchPanel extends React.Component<P, {}> {
...
   }
 
 
   componentDidMount() {
+    const { store: { getState } } = this.props;
+    const { btnSubmitEnabled } = this.state;
     const id = findParam('id') || findParam('savedId') || findParam('recordid');
+    if (!btnSubmitEnabled) {
+      const form = getState().form.searchForm;
+      if (typeof (form) !== 'undefined' && typeof (form.values) !== 'undefined' && typeof (form.values.searchTextArea) !== 'undefined') {
```
Comment (ryandberger) [inline src/components/Search/Panel/SearchPanel.js:68]: Can just say `form?.values?.searchTextArea?.length` to reduce these 2 lines down to 1?

File: src/components/Cataloguing/Marc/Record/index.js (hunk 1 above)
Comment (EstebanSM) [reply src/components/Cataloguing/Marc/Record/index.js]: ok, I have changed the code using string interpolation

File: src/components/Cataloguing/Marc/Record/index.js (hunk 2 above)
Comment (EstebanSM) [reply src/components/Cataloguing/Marc/Record/index.js]: Now, first I ask about the edit mode and then for other conditions in an else

File: src/components/Search/Panel/SearchPanel.js (hunk 3 above)
Comment (EstebanSM) [reply src/components/Search/Panel/SearchPanel.js]: I didn't know that possibility to simplify, thanks

**Review by ryandberger — APPROVED** [review]
Looks good now, thanks for making the suggested changes!

**Pull request merged**